### PR TITLE
Fix l3-dependent L4/L7 rules applying to CIDR egress traffic

### PR DESCRIPTION
**Pull request by joestringer**
Fixes: #3414

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

File: bpf/lib/l4.h [hunk 1]
```diff
@@ -135,26 +135,51 @@ struct l4_allow
...
 
+#ifdef CFG_L4_EGRESS
 	BPF_L4_MAP(allowed, dport, nexthdr, CFG_L4_EGRESS);
+#else
+	/* If there are only l3-dependent L4 rules, then CIDR-based L4 is
+	 * completely wildcarded */
+	allowed = 0;
+#endif /* CFG_L4_EGRESS */
+#ifdef CFG_L3L4__EGRESS
+	if (l3dependent) {
```
Comment (tgraf) [inline bpf/lib/l4.h]: Let's reset allowed and treat each call to `BPF_L4_MAP()` independently.

**Review by tgraf — CHANGES_REQUESTED** [review]

**Comment by joestringer** [comment]
test-me-please

File: bpf/lib/l4.h (hunk 1 above)
Comment (joestringer) [reply bpf/lib/l4.h]: Done, separated out into entirely different invocations.

**Review by tgraf — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Review by eloycoto — APPROVED** [review]

**Comment by joestringer** [comment]
L3/L4 checks failed in CI but passed locally. I'm running all of the policies tests locally to see if anything crops up.

**Comment by joestringer** [comment]
test-me-please

**Pull request merged**